ledger-service/http-json/src/main/scala/com/digitalasset/http/util/LedgerOffsetUtil.scala

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.http.util
+
+import com.digitalasset.ledger.api.v1.ledger_offset.LedgerOffset
+import com.digitalasset.ledger.api.v1.ledger_offset.LedgerOffset.Value
+
+object LedgerOffsetUtil {
+  implicit val AbsoluteOffsetOrdering: Ordering[LedgerOffset.Value.Absolute] =
+    new Ordering[LedgerOffset.Value.Absolute] {
+      override def compare(x: Value.Absolute, y: Value.Absolute): Int =
+        if (isCompositeOffsetFormat(x)) CompositeAbsoluteOffsetIntOrdering.compare(x, y)
+        else IntAbsoluteOffsetOrdering.compare(x, y)
+    }
+
+  private val IntAbsoluteOffsetOrdering: Ordering[LedgerOffset.Value.Absolute] =
+    Ordering.by[LedgerOffset.Value.Absolute, Long](_.value.toLong)
+
+  private val CompositeAbsoluteOffsetIntOrdering: Ordering[LedgerOffset.Value.Absolute] =
+    Ordering.by[LedgerOffset.Value.Absolute, (Long, Long)](parseCompositeOffset)
+
+  private def isCompositeOffsetFormat(a: LedgerOffset.Value.Absolute): Boolean =
+    a.value.contains('-')
+
+  private def parseCompositeOffset(a: LedgerOffset.Value.Absolute): (Long, Long) = {
```

Comment:
```suggestion
  private def parseOffset(a: LedgerOffset.Value.Absolute): Long \/ (Long, Long) = {
```

with success in the `case Array(n) =>` case. Then you can do away with the `isCompositeOffsetFormat` and derive the order from the order of `Long \/ (Long, Long)`.

Reply:
I don't like this... this assume that we would be comparing Numberic and Composite offsets, which should never happen, if happens this is an error! Which means that offsets are coming from two different ledgers.

Reply:
Principle of explosion: if something happens that should never happen, then anything is fine, right? (In this case you'll just get either the error from the server about start > end, or empty stream.)

Reply:
To put it another way: when you compare two `Long`s that should never be negative, you tend not to assert there that they are `>= 0`, you just assume it (and there happens to be total behavior in the negative case anyway).

Reply:
To be honest, I should add that the main goal was to eliminate the blind boolean implied by pairing a boolean test with a function that presumes the boolean test succeeded; making the comparison total was just to simplify the code (and testing thereof) further.